File: Evolutionary_flow/TrojanEncoder/trojanEncoder.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

char *mat[102] ={
"000000", "ADD",        //0
"000001", "ADDI",       //1
"000010", "ADDIU",      //2
"000011", "ADDU",       //3
"000100", "AND",        //4
"000101", "ANDI",       //5
"000110", "BEQ",        //6
"000111", "BGEZ",       //7
"100011", "BGEZAL",     //8
"001001", "BGTZ",       //9
"001010", "BLEZ",       //10
"001011", "BLTZ",       //11
"101000", "BLTZAL",     //12
"001101", "BNE",        //13
"001110", "BREAK",      //14
"001111", "COP0",       //15
"010000", "J",          //16
"010001", "JAL",        //17
"010010", "JALR",       //18
"010011", "JR",         //19
"010100", "LUI",        //20
"010101", "LW",         //21
"010110", "LWC0",       //22
"010111", "MFC0",       //23
"011000", "MFHI",       //24
"011001", "MFLO",       //25
"011010", "MTC0",       //26
"011011", "MTHI",       //27
"011100", "MTLO",       //28
"011101", "MULT",       //29
"011110", "MULTU",      //30
"011111", "NOR",        //31
"100000", "OR",         //32
"100001", "ORI",        //33
"100010", "SLL",        //34
"100011", "SLLV",       //35
"100100", "SLT",        //36
"100101", "SLTI",       //37
"100110", "SLTIU",      //38
"100111", "SLTU",       //39
"101000", "SRA",        //40
"101001", "SRAV",       //41
"101010", "SRL",        //42
"101011", "SRLV",       //43
"101100", "SUB",        //44
"101101", "SUBU",       //45
"101110", "SW",         //46
"101111", "SWC0",       //47
"110000", "SYSC",       //48
"110001", "XOR",        //49
"110010", "XORI"        //50
};

int main(int argc, char ** argv)
{
	int i ;

	printf("instructions = {\n");

	for(i = 0; i < 102; i += 2)
		printf("'%s'\t\t: '%s',\t\t#%d\n", mat[i+1], mat[i], i/2);

	printf("}\n");
	return 0;
}

File: Evolutionary_flow/TrojanEncoder/trojanEncoder.py
import sys

instructions = {
'ADD'		: '000000',		#0
'ADDI'		: '000001',		#1
'ADDIU'		: '000010',		#2
'ADDU'		: '000011',		#3
'AND'		: '000100',		#4
'ANDI'		: '000101',		#5
'BEQ'		: '000110',		#6
'BGEZ'		: '000111',		#7
'BGEZAL'	: '100011',		#8
'BGTZ'		: '001001',		#9
'BLEZ'		: '001010',		#10
'BLTZ'		: '001011',		#11
'BLTZAL'	: '101000',		#12
'BNE'		: '001101',		#13
'BREAK'		: '001110',		#14
'COP0'		: '001111',		#15
'J'		    : '010000',		#16
'JAL'		: '010001',		#17
'JALR'		: '010010',		#18
'JR'		: '010011',		#19
'LUI'		: '010100',		#20
'LW'		: '010101',		#21
'LWC0'		: '010110',		#22
'MFC0'		: '010111',		#23
'MFHI'		: '011000',		#24
'MFLO'		: '011001',		#25
'MTC0'		: '011010',		#26
'MTHI'		: '011011',		#27
'MTLO'		: '011100',		#28
'MULT'		: '011101',		#29
'MULTU'		: '011110',		#30
'NOR'		: '011111',		#31
'OR'		: '100000',		#32
'ORI'		: '100001',		#33
'SLL'		: '100010',		#34
'SLLV'		: '100011',		#35
'SLT'		: '100100',		#36
'SLTI'		: '100101',		#37
'SLTIU'		: '100110',		#38
'SLTU'		: '100111',		#39
'SRA'		: '101000',		#40
'SRAV'		: '101001',		#41
'SRL'		: '101010',		#42
'SRLV'		: '101011',		#43
'SUB'		: '101100',		#44
'SUBU'		: '101101',		#45
'SW'		: '101110',		#46
'SWC0'		: '101111',		#47
'SYSC'		: '110000',		#48
'XOR'		: '110001',		#49
'XORI'		: '110010',		#50
}

for instr in sys.argv[1:]:
    print(instr + " : "  + instructions[instr])
